extensions/SharkPool/BetterInput.js

```diff
@@ -0,0 +1,1305 @@
+          this.cancelButtonColor = colorValue;
+          break;
+        case "Button 3":
+          this.button3Color = colorValue;
+          break;
+        case "Button 4":
+          this.button4Color = colorValue;
+          break;
+        case "Dropdown Buttons":
+          this.optionbuttonColor = colorValue;
+          break;
+        case "Slider":
+          this.sliderColor = colorValue;
+          break;
+        case "Button 1 Text":
+          this.submitButtonTextColor = colorValue;
+          break;
+        case "Button 2 Text":
+          this.cancelButtonTextColor = colorValue;
+          break;
+        case "Button 3 Text":
+          this.button3TextColor = colorValue;
+          break;
+        case "Button 4 Text":
+          this.button4TextColor = colorValue;
+          break;
+        case "Dropdown Text":
+          this.optionbuttonTextColor = colorValue;
+          break;
+      }
+    }
+
+    setBorderRadius(args) {
+      const element = args.ELEMENT;
+      let value = args.VALUE;
+
+      if (value < 0) {
+        value = 0;
+      }
+
+      switch (element) {
+        case "Textbox":
+          this.textBoxBorderRadius = value;
+          break;
+        case "Button 1":
+          this.cancelButtonBorderRadius = value;
+          break;
+        case "Button 2":
+          this.submitButtonBorderRadius = value;
+          break;
+        case "Button 3":
+          this.button3BorderRadius = value;
+          break;
+        case "Button 4":
+          this.button4BorderRadius = value;
+          break;
+        case "Input Box":
+          this.inputBoxRadius = value;
+          break;
+        case "Dropdown Buttons":
+          this.optionbuttonBorderRadius = value;
+          break;
+        case "Slider":
+          this.sliderBorderRadius = value;
+          break;
+      }
+    }
+
+    setEnterEffect(args) {
+      const enterEffect = args.ENTER_EFFECT;
+      switch (enterEffect) {
+        case "None":
+          this.enterEffect = null;
+          break;
+        case "Fade in":
+          this.enterEffect = "fadeIn";
+          break;
+        case "Grow":
+          this.enterEffect = "growIn";
+          break;
+        case "Both":
+          this.enterEffect = "Both";
+          break;
+      }
+    }
+
+    setExitEffect(args) {
+      const exitEffect = args.EXIT_EFFECT;
+      switch (exitEffect) {
+        case "None":
+          this.exitEffect = null;
+          break;
+        case "Fade out":
+          this.exitEffect = "fadeOut";
+          break;
+        case "Shrink":
+          this.exitEffect = "shrinkOut";
+          break;
+        case "Both":
+          this.exitEffect = "Both";
+          break;
+      }
+    }
+
+    setPrePosition(args) {
+      this.textBoxX = args.X;
+      this.textBoxY = args.Y * -1;
+    }
+
+    setPosition(args) {
+      this.textBoxX = args.X;
+      this.textBoxY = args.Y * -1;
+
+      const overlays = document.querySelectorAll(".ask-box");
```

Comment:
querySelectorAll is slow and prone to collisions. it would be best if you could track those elements yourself

Reply:
Fixed

Reply:
You still use querySelectorAll in removeAskBoxes